src/openms/source/QC/DBSuitability.cpp

```diff
@@ -101,102 +91,51 @@ namespace OpenMS
-        continue;
-      }
-
-      // second hit is db hit
-      ++data.num_interest;
-
-      // check for re-ranking
-      if (no_rerank)
-      {
-        ++data.num_top_novo;
-        continue;
-      }
-
-      // check for xcorr score
-      if (!top_hit.metaValueExists("MS:1002252") || !second_hit->metaValueExists("MS:1002252"))
-      {
-        throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No cross correlation score found at peptide hit. Only Comet search engine is supported right now."));
-      }
-
-      double top_xscore_mw = double(top_hit.getMetaValue("MS:1002252")) / top_hit.getSequence().getMonoWeight();
-      double second_xscore_mw = double(second_hit->getMetaValue("MS:1002252")) / second_hit->getSequence().getMonoWeight();
-      if (top_xscore_mw - second_xscore_mw <= data.cut_off)
-      {
-        ++data.num_top_db;
-        ++data.num_re_ranked;
-      }
-      else
-      {
-        ++data.num_top_novo;
-      }
-    }
-
-    if (data.num_top_db == 0 && data.num_top_novo == 0)
-    {
-      OPENMS_LOG_WARN << "Identifications could not be assigned to either the database or the deNovo protein. Probably your FDR threshold is too strict." << endl;
-      data.suitability = DBL_MAX;
-      return;
-    }
+    // calculate suitability
+    SuitabilityData d;
+    results_.push_back(d);
+    SuitabilityData& data = results_.back();
+    calculateSuitability_(pep_ids, data);
 
-    data.suitability = double(data.num_top_db) / (data.num_top_db + data.num_top_novo);
+    // calculate correction of suitability with extrapolation
+    
+    // sampled run
+    // maybe multiple runs? could be controlled with a parameter
+    double ratio = 0.5;
+    vector<FASTAFile::FASTAEntry> sampled_db = getSubsampledFasta_(original_fasta, ratio);
+    sampled_db.insert(sampled_db.end(), novo_fasta.begin(), novo_fasta.end());
+    calculateDecoys_(sampled_db);
+    vector<PeptideIdentification> subsampled_ids = runIdentificationSearch_(exp, sampled_db, search_info.first, search_info.second);
+
+    SuitabilityData sampled_data;
+    calculateSuitability_(subsampled_ids, sampled_data);
+
+    // slopes of db and deNovo hits
+    double db_slope = (int(sampled_data.num_top_db) - int(data.num_top_db)) / (-ratio);
+    double deNovo_slope = (int(sampled_data.num_top_novo) - int(data.num_top_novo)) / (-ratio);
+
+    // calculate deNovo intercept (maximum deNovo ids)
+    calculateDecoys_(novo_fasta);
+    Int deNovo_intercept = countIdentifications_(runIdentificationSearch_(exp, novo_fasta, search_info.first, search_info.second));
+    // db_intercept is estimated to be 0
+
+    // calculate correction factor
+    double target_ratio = (-deNovo_intercept) / deNovo_slope; // ratio that corresponds to suitability of 1
+    double db_hits_at_ratio = db_slope * target_ratio; // db hits for suitability 1
+    double factor = db_hits_at_ratio / deNovo_intercept;
+
+    data.corr_factor = factor;
```

Comment:
put these three lines which compute corrected factors into a member of `data`, e.g. `setCorrectionFactor(double factor)`